eng/pipelines/templates/steps/dependency.tests.yml

```diff
@@ -0,0 +1,47 @@
+parameters:
+- name: ProjectFilesOutputFolder
+  type: string
+  default: "$(Build.ArtifactStagingDirectory)"
+- name: NumOfTestsPerJob
+  type: number
+  default: 20
+- name: OutputProjectFilePath
+  type: string
+  default: "$(System.DefaultWorkingDirectory)/projects.txt"
+- name: TestDependsOnDependency
+  type: string
+  default: "not-specific"
+- name: ProjectListOverrideFileProperty
+  type: string
+  default: ''
+- name: MatrixConfigs
+  type: string
+  default: './eng/pipelines/templates/stages/platform-matrix.json'
+steps:
+  - pwsh: |
+      dotnet build /t:ProjectDependsOn ./eng/service.proj `
+        /p:TestDependsOnDependency="${{parameters.TestDependsOnDependency}}" `
+        /p:IncludeSrc=false /p:IncludeStress=false /p:IncludeSamples=false  `
+        /p:IncludePerf=false /p:RunApiCompat=false `
+        /p:InheritDocEnabled=false /p:BuildProjectReferences=false `
+        /p:OutputProjectFilePath="${{parameters.OutputProjectFilePath}}"
+    displayName: Find all test dependencies
+
+  - task: PowerShell@2
+    inputs:
+      targetType: filePath
+      filePath: eng/scripts/Generate-Dependency-Test-References.ps1
+      arguments: >-
+        -ProjectListFilePath ${{ parameters.OutputProjectFilePath }}
+        -ProjectFilesOutputFolder ${{ parameters.ProjectFilesOutputFolder }}
+        -NumOfTestsPerJob ${{ parameters.NumOfTestsPerJob }}
+        -MatrixConfigsFile ${{parameters.MatrixConfigs}}
```

Comment:
I think it would be better to pass in the `MatrixConfigs` and `AdditionalMatrixConfigs` jsonified object from the outer parameter context, that way we can support the existing way matrix configs are specified rather than hardcoding the path in multiple places in a way that's not overridable.

Reply:
Done.